tests/test_run_plan.py

```diff
@@ -118,6 +123,41 @@ class TestFullRunPlan:
     run_plan = RunPlan.create(
         all_known_projects={execution_1.project, execution_2.project}, plan=full_plan
     )
+    test_run_json = Path("tests/.mpyl/run_plan.json")
+
+    @patch("src.mpyl.run_plan.RUN_PLAN_JSON_FILE", test_run_json)
+    def test_write_to_json(self):
+        full_plan = {
+            build_stage: {execution_1, execution_2},
+            test_stage: {execution_1},
+            deploy_stage: {execution_2, cached_execution},
+        }
+        run_plan = RunPlan.create(
+            all_known_projects={execution_1.project},
+            plan=full_plan,
+        )
+
+        run_plan.write_to_json_file()
+        with open(self.test_run_json, encoding="utf-8") as file:
+            run_plan_json = json.load(file)
+            project1 = next(
+                project
+                for project in run_plan_json
+                if project["service"] == "project 1"
+            )
+            project2 = next(
+                project
+                for project in run_plan_json
+                if project["service"] == "project 2"
+            )
+            assert project1["build"] is True
+            assert project1["test"] is True
+            assert project1["deploy"] is False
+            assert project2["build"] is True
+            assert project2["test"] is False
+            assert project2["deploy"] is True
```

Comment:
We should also test at least that `cached_execution` gets written properly:

```python
with open(self.test_run_json, encoding="utf-8") as file:
  run_plan_json = json.load(file)
  
  def get_project_json(name: str):
    return next(
      (project for project in run_plan_json if project["service"] == name),
      None,
    )

  project_1 = get_project_json("project 1")
  …
   
  project_2 = get_project_json("project 2")
  …

  cached_project = get_project_json("a cached project")
  assert cached_project["build"] is False
  assert cached_project["test"] is False
  assert cached_project["deploy"] is False

  assert get_project_json("an unknown project") is None
```

Reply:
We should also consider adding this same functionality in the other test suites (after selection, etc). Not sure how much it adds though.

Reply:
Both are done!